src/karthik/group-people-data/groupPeopleData.spec.ts

```diff
@@ -0,0 +1,34 @@
+import { groupPeopleByCity } from './groupPeopleData';
+import {
+  testData,
+  expectedResult,
+  partialTestData,
+  partialExpectedResult
+} from './data';
+
+describe('testing getCityBasedPeopleGroup function', () => {
+  it('testing if testData returns the expectedResult', () => {
+    const result = groupPeopleByCity(testData);
+    expect(result).toEqual(expectedResult);
+  });
+
+  it('testing if partialTestData does not return the expectedResult', () => {
+    const result = groupPeopleByCity(partialTestData);
+    expect(result).not.toEqual(expectedResult);
+  });
+
+  it('testing if partialTestData returns the partialExpectedResult', () => {
+    const result = groupPeopleByCity(partialTestData);
+    expect(result).toEqual(partialExpectedResult);
+  });
+
+  it('testing if testData does not return the partialExpectedResult', () => {
+    const result = groupPeopleByCity(testData);
+    expect(result).not.toEqual(partialExpectedResult);
```

Comment:
I think the below tests are redundant in this case, since the test cases are already placed to check the respective outputs for the given inputs, no need to cross verify them with the same arrangement against with (`not equal to`) opposite outputs.
- `testing if partialTestData does not return the expectedResult`
- `testing if testData does not return the partialExpectedResult`

Reply:
fixed